README.md

```diff
@@ -4,175 +4,225 @@
+an [`AccessTokenRequest`](https://docs.rs/dcaf/latest/dcaf/endpoints/token_req/struct.AccessTokenRequest.html), which
+has to include at least a
+`client_id`. We'll also specify an audience, a scope (
+using [`TextEncodedScope`](https://docs.rs/dcaf/latest/dcaf/common/scope/struct.TextEncodedScope.html)---note that
+[binary-encoded scopes](https://docs.rs/dcaf/latest/dcaf/common/scope/struct.BinaryEncodedScope.html)
+or [AIF-encoded scopes](https://docs.rs/dcaf/latest/dcaf/common/scope/struct.AifEncodedScope.html) would also work), as
+well as a
+[`ProofOfPossessionKey`](https://docs.rs/dcaf/latest/dcaf/common/cbor_values/enum.ProofOfPossessionKey.html) (the key
+the access token should be bound to) in the `req_cnf` field.
 
 Creating, serializing and then de-serializing such a structure would look like this:
+
 ```rust
 use dcaf::{AccessTokenRequest, ToCborMap, ProofOfPossessionKey, TextEncodedScope};
 
 let request = AccessTokenRequest::builder()
-   .client_id("myclient")
-   .audience("valve242")
-   .scope(TextEncodedScope::try_from("read")?)
-   .req_cnf(ProofOfPossessionKey::KeyId(base64::decode("6kg0dXJM13U")?))
-   .build()?;
+.client_id("myclient")
+.audience("valve242")
+.scope(TextEncodedScope::try_from("read") ? )
+.req_cnf(ProofOfPossessionKey::KeyId(base64::decode("6kg0dXJM13U") ? ))
+.build() ?;
 let mut encoded = Vec::new();
-request.clone().serialize_into(&mut encoded)?;
+request.clone().serialize_into( & mut encoded) ?;
 assert_eq!(AccessTokenRequest::deserialize_from(encoded.as_slice())?, request);
 ```
 
-### Access Tokens
+## Access Tokens
+
+Following up from the previous example, let's assume we now want to create a signed
+access token containing the existing `key`, as well as claims about the audience and issuer
+of the token, using an existing cipher of type `FakeCrypto`[^cipher]:
 
-Following up from the previous example, let's assume we now want to create a signed access token containing the
-existing `key`, as well as claims about the audience and issuer of the token, using an existing `cipher`[^cipher]:
 ```rust
-use dcaf::{ToCborMap, sign_access_token, verify_access_token};
-use coset::cwt::ClaimsSetBuilder;
-use coset::Header;
-use coset::iana::CwtClaimName;
 
+# [derive(Clone)]
+
+let rng = FakeRng;
+let key = FakeKey { key: [1, 2, 3, 4, 5], kid: [0xDC, 0xAF]};
+let cose_key: CoseKey = key.to_cose_key();
 let claims = ClaimsSetBuilder::new()
-   .audience("valve242".to_string())
-   .claim(CwtClaimName::Cnf, key.to_ciborium_value())
-   .claim(CwtClaimName::Scope, Value::Text("read".to_string()))
-   .build();
-let token = sign_access_token(claims, &mut cipher, None, None, None)?;
-assert!(verify_access_token(&token, &mut cipher, None).is_ok());
+.audience(String::from("coaps://rs.example.com"))
+.issuer(String::from("coaps://as.example.com"))
+.claim(CwtClaimName::Cnf, cose_key.to_cbor_value() ? )
+.build();
+let token = sign_access_token::<FakeCrypto, FakeRng>( & key, claims, None, None, None, rng) ?;
```

Comment:
Same here.

Reply:
Fixed in bdcb6ad917e5b63c51434feb6d449aed1b14d830.